packages/hash/frontend/src/components/layout/PageHeader/SearchBar.tsx

```diff
@@ -97,6 +97,10 @@ export const SearchBar: React.VFC = () => {
   // present loading screen while waiting for the user to stop typing
   const isLoading = loading || displayedQuery !== submittedQuery;
 
+  // NOTE: This component is currently returning an empty div as a way to disable it when opensearch is off
+  if (process.env.NEXT_PUBLIC_HASH_OPENSEARCH_ENABLED !== "true") {
+    return <div />;
+  }
```

Comment:
We may end up with a situation some component logic is broken because the server is off. So instead of conditionally returning an empty div, we can make the entire component conditional:

```tsx
const SearchBarWhenSearchIsEnabled = ...;

const SearchBarWhenSearchIsDisabled: VoidFunctionComponent = () => {
  return <div />
}

export const SearchBar = process.env.NEXT_PUBLIC_HASH_OPENSEARCH_ENABLED === "true"
  ? SearchBarWhenSearchIsEnabled
  : SearchBarWhenSearchIsDisabled;
```

We [did something similar for `useCollabPositions()` hook](https://github.com/hashintel/hash/commit/b3aae3e0895f1745e213add9e1c8ee40257daaea#diff-4b88e35f8fb69be9809c05199458282ba9191290039868af762837809acc120aR92-R99).